src/main/java/org/folio/rest/impl/LedgersApi.java

```diff
@@ -39,14 +44,30 @@ public void postFinanceLedgers(String lang, Ledger entity, Map<String, String> h
 
   @Validate
   @Override
-  public void getFinanceLedgers(int offset, int limit, String query, String lang, Map<String, String> headers,
-      Handler<AsyncResult<Response>> handler, Context ctx) {
+  public void getFinanceLedgers(String fiscalYear, int offset, int limit, String query, String lang, Map<String, String> headers,
+     Handler<AsyncResult<Response>> handler, Context ctx) {
 
     LedgersHelper helper = new LedgersHelper(headers, ctx, lang);
 
-    helper.getLedgers(limit, offset, query)
-      .thenAccept(types -> handler.handle(succeededFuture(helper.buildOkResponse(types))))
-      .exceptionally(fail -> handleErrorResponse(handler, helper, fail));
+    if(Objects.isNull(fiscalYear)) {
+      helper.getLedgers(limit, offset, query)
+        .thenAccept(ledgersCollection -> handler.handle(succeededFuture(helper.buildOkResponse(ledgersCollection))))
+        .exceptionally(fail -> handleErrorResponse(handler, helper, fail));
+    } else {
+      helper.getLedgers(limit, offset, query)
+        .thenCombine(helper.getLedgerFYsByFiscalYearId(fiscalYear), (ledgerCollection, ledgerFYCollection) -> {
+          Map<String, LedgerFY> ledgerFYGroupedByLedgerId = StreamEx.of(ledgerFYCollection.getLedgerFY())
+            .filter(ledgerFY -> ledgerFY.getFiscalYearId().equals(fiscalYear))
```

Comment:
This filter is not needed here because the `getLedgerFYsByFiscalYearId` returns `ledgerFY` records only with `fiscalYearId==fiscalYear`.

Reply:
As far as I understand multicolumn index of ledgerId and fiscalYearId should be unique according to schema https://github.com/folio-org/mod-finance-storage/blob/47b4c29c800224241fa20a0b86c318e96672f23b/src/main/resources/templates/db_scripts/schema.json#L368. A multicolumn unique index will only reject cases where all indexed columns are equal in multiple rows. This means that potentially we can have in this map the following combinations: fiscalYear_1-ledger_1, fiscalYear_1-ledger_2, etc. I used filter here just to prevent this and to reduce number of records.